galadriel/memory/memory_repository.py

```diff
@@ -20,23 +20,26 @@ class MemoryRepository:
 
     def __init__(
         self,
-        api_key: str,
         short_term_memory_limit: int = 20,
         embedding_model: Optional[str] = "text-embedding-3-large",
         agent_name: Optional[str] = "agent",
+        use_long_term_memory: bool = False,
```

Comment:
No need to have additional `use_long_term_memory` flag. I'd follow GPTs idea of setting up vector_store if api_key and embedding_model are passed, and then in other functions check if vectore_store exists. Like here https://github.com/galadriel-ai/galadriel/commit/f4331ab947c32939897f43812c371d601def6402#diff-ecea9e03370293d74e61094f622b6fb21ea45b9f58c5948a369ddd8c167604b0R43